javascripts/content_scripts/posting_button.js

```diff
@@ -1,131 +1,424 @@
-/*global privlyUrlReceiptNode:false, pendingPost:false, chrome:false, ls:true,  */
+/*global privlyPosting:false, chrome:false, ls:true,  */
 
 /**
- * Shows a button in editable content areas for starting a Privly Post.
+ * This content-script provides Privly Posting Button feature.
+ * It listens events of editable elements (textarea / contentEditable elements)
+ * and create or show the posting button.
+ *
+ * The button is created as a sibling of the editable element, which enables
+ * it to share the same parent with the editable element. This gives it better
+ * positioning in most of situations. It works well even when the parent has
+ * `position:fixed` style.
  */
-function addPrivlyButton() {
 
-  // Create a div to handle the privly button
-  var div = document.createElement("div");
+(function() {
```

Comment:
You will likely need to refactor this before you will be able to unit test it since the members will not be accessible to the testing scope.